concent_api/core/tests/test_integration_subtask_results_verify.py

```diff
@@ -0,0 +1,228 @@
+            message_type=message.concents.AckSubtaskResultsVerify,
+            fields={
+                'subtask_results_verify': self._prepare_subtask_results_verify(serialized_subtask_results_verify),
+                'file_transfer_token': self._prepare_file_transfer_token(subtask_results_verify_time_str),
+            }
+        )
+        self._assert_stored_message_counter_increased(increased_by=2)
+        send_verification_request_mock.assert_called_once()  # TODO: once verification_request implemented change to 'assert_called_once_with'
+
+    def _prepare_subtask_results_verify(self, serialized_subtask_results_verify):
+        subtask_results_verify = load(
+            serialized_subtask_results_verify,
+            CONCENT_PRIVATE_KEY,
+            self.PROVIDER_PUBLIC_KEY,
+            check_time=False,
+        )
+        subtask_results_verify.encrypted = False
+        return subtask_results_verify
+
+    def _prepare_file_transfer_token(self, subtask_results_verify_time_str):
+        with freeze_time(subtask_results_verify_time_str):
+            file_transfer_token = create_file_transfer_token(
+                self.report_computed_task,
+                b64encode(decode_key(self._get_encoded_provider_public_key())),
+                "upload"
+            )
+            file_transfer_token.encrypted = False
+        return file_transfer_token
+
+    def _create_serialized_subtask_results_verify(
+        self,
+        reason_of_rejection=message.tasks.SubtaskResultsRejected.REASON.VerificationNegative
+    ):
+        subtask_results_verify, subtask_results_verify_time_str = self._create_subtask_results_verify(
+            reason_of_rejection)
+        serialized_subtask_results_verify = self._get_serialized_subtask_results_verify(
+            subtask_results_verify=subtask_results_verify
+        )
+        return serialized_subtask_results_verify, subtask_results_verify_time_str
+
+    def _create_subtask_results_verify(
+        self,
+        reason_of_rejection=message.tasks.SubtaskResultsRejected.REASON.VerificationNegative
+    ):
+        subtask_result_rejected_time_str = "2018-04-01 10:30:00"
+        subtask_results_rejected = self._get_deserialized_subtask_results_rejected(
+            reason=reason_of_rejection,
+            timestamp=subtask_result_rejected_time_str,
+            report_computed_task=self.report_computed_task)
+        subtask_results_verify_time_str = self._add_time_offset_to_date(subtask_result_rejected_time_str,
+                                                                        settings.ADDITIONAL_VERIFICATION_CALL_TIME / 2)
+        subtask_results_verify = self._get_deserialized_subtask_results_verify(
+            timestamp=subtask_results_verify_time_str,
+            subtask_results_rejected=subtask_results_rejected)
+        return subtask_results_verify, subtask_results_verify_time_str
+
+    def _create_report_computed_task(self):
```

Comment:
I think there is a function in test utils which already does that.

Reply:
I am using it in my function - here `_create_report_computed_task` is used for easier setup